gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3039,6 +3037,16 @@ func checkIntegerType(xt Type) {
 	}
 }
 
+// like checkOrConvertType() but for bool type.
+func checkOrConvertBoolType(store Store, last BlockNode, x Expr) {
+	if cx, ok := x.(*ConstExpr); ok {
+		convertConst(store, last, cx, BoolType)
+	} else if x != nil {
+		t := evalStaticTypeOf(store, last, x)
+		checkType(baseOf(t), BoolType, false)
```

Comment:
checkOrConvertIntegerType calls checkOrConvertType here, I wonder why there is asymmetry, and if we need it.

Reply:
I see checkOrConvertIntegerType calling checkIntegerType here https://github.com/gnolang/gno/blob/80c9ddc00bd3a0c6a401d06968ce17ab904c97ff/gnovm/pkg/gnolang/preprocess.go#L3022

Reply:
> also take a look at this, which I haven't reviewed; it makes changes to dependent functions. #1426

I think `checkType` would have to be changed to `assertAssignableTo` after that PR merges. I'll add a comment